text/0000-target-tier-policy.md

```diff
@@ -0,0 +1,222 @@
+
+- No central decision is required to add a new tier 3 target. Reviewers may
+  always use their own best judgment regarding the quality of work, and the
+  suitability of a target for the Rust project.
+- If a reviewer wishes to consult a broader team for additional guidance, they
+  should contact the compiler team.
+- If a proposed target or target-specific patch substantially changes code
+  shared with other targets (not just target-specific code), the reviewer
+  should consult the compiler team.
+- If the proposer of a target wishes to appeal the rejection of a target, they
+  may contact the compiler team.
+- Tier 3 targets must use naming consistent with any existing targets; for
+  instance, a target for a similar CPU or OS should not gratuitously use an
+  inconsistent name for that CPU or OS. Targets should normally use the same
+  names as used elsewhere in the broader ecosystem (such as in other
+  toolchains), unless they have a very good reason to diverge.
+- Tier 3 targets may have unusual requirements to build or use, but must not
+  create legal issues for the Rust project or for developers who work on those
+  targets.
+- Where possible, tier 3 targets may wish to provide documentation for the Rust
+  community for how to build and run tests for the platform, ideally using
+  emulation.
+- Tier 3 targets must not impose burden on the authors of pull requests, or
+  other developers in the community, to maintain the target. In particular,
+  do not post comments (automated or manual) on a PR that suggests a block on
+  the PR based on the target. (A PR author may choose to help with a tier 3
+  target but is not required to.)
+- Patches adding or updating tier 3 targets must not break any existing target.
+- If a tier 3 target shows no signs of activity and has not built for some
+  time, and removing it would improve the quality of the Rust codebase, we may
+  post a PR to remove it; any such PR will be CCed to people who have
+  previously worked on the platform, to check potential interest.
+
+## Tier 2 target policy
+
+At this tier, the Rust project guarantees that a target builds, and will reject
+patches that fail to build on a target. Thus, we place requirements that ensure
+the target will not block forward progress of the Rust project.
+
+- Any new tier 2 target requires compiler team approval based on these
+  requirements.
+- In addition, the infrastructure team must approve the integration of the
+  target into CI, and the CI-related requirements. This review and approval
+  will typically take place in the PR adding the target to CI.
+- A tier 2 target must have value to people other than its maintainers.
+- Any new tier 2 target must have a designated team of developers on call to
+  consult on target-specific build-breaking issues, or if necessary to develop
+  target-specific language or library implementation details. (This team should
+  in almost all cases have at least 2 developers.)
```

Comment:
And if it doesn't have two developers? This seems like it shouldn't be part of a *requirements* list.

Reply:
Fixed. I had intended for this to be a soft requirement, but on further reflection, I think it's reasonable to make this a hard requirement. I don't think we should have a tier 2 target with only a single maintainer.